DOC-7628 correct Analytics dataset prerequisites

osfameron opened the pull request:
* update creation of dataset as per
   https://blog.couchbase.com/collections-simplify-your-analysis-with-couchbase-analytics/

 ** this now creates the dataset as "`bucket`.`scope`" instead of "`bucket.scope`"
 ** as the convenience scope.analyticsQuery call seemed to work previously in the
    Java SDK, this suggests the previous behaviour was incorrect?
    Now, the handle-scope example (which previously ran) gives the error:
       `com.couchbase.client.core.error.DataverseNotFoundException: The analytics dataverse is not found`

 ** Commented out that section of code and accompanying documentation for now.
 ** will also raise a bug/clarification ticket with sdk team

osfameron committed:
DOC-7628 correct Analytics dataset prerequisites

 * update creation of dataset as per
   https://blog.couchbase.com/collections-simplify-your-analysis-with-couchbase-analytics/

 ** this now creates the dataset as "`bucket`.`scope`" instead of "`bucket.scope`"
 ** as the convenience scope.analyticsQuery call seemed to work previously in the
    Java SDK, this suggests the previous behaviour was incorrect?
    Now, the handle-scope example (which previously ran) gives the error:
       `com.couchbase.client.core.error.DataverseNotFoundException: The analytics dataverse is not found`

 ** Commented out that section of code and accompanying documentation for now.
 ** will also raise a bug/clarification ticket with sdk team

maria-robobug reviewed (approved):
LGTM, 👍

maria-robobug commented on modules/howtos/pages/analytics-using-sdk.adoc:
```diff
@@ -205,31 +205,26 @@ In addition to creating a dataset with a WHERE clause to filter the results to d
-CREATE DATASET `airports-collection` ON `travel-sample`.inventory.airport;
--- compare with:
--- CREATE DATASET airports ON `travel-sample` WHERE `type` = "airport";
+ALTER COLLECTION `travel-sample`.inventory.airport ENABLE ANALYTICS;
```
Didn't know about this way to do it, nice! 👍

osfameron committed:
DOC-7628 re-enable commented out scopes part

The SDK code now works fine, reenabling the sample code and doc,
and closing JCBC-1788

osfameron commented:
Looks like JCBC-1788 is resolved, so reenabling the commented out section, this should now be good to merge?

RichardSmedley reviewed (approved):
LGTM  :+1:
